src/Connector.js

```diff
@@ -347,7 +355,12 @@ class Connector {
 
   async publishData(id, dataList) {
     const promises = dataList.map(async (data) => {
-      await this.client.publish(`/${id}/${data.sensorId}/attrs/value`, data.value.toString());
+      let value = data.value.toString();
+      if (isBase64(data.value)) {
+        value = Buffer.from(data.value, 'base64').toString('utf-8');
+      }
+
+      await this.client.publish(`/${id}/${data.sensorId}/attrs/value`, value);
     });
 
     await Promise.all(promises);
```

Comment:
The `Promise.all` waits for an array of promises but it is an array of undefined